src/viam/app/ml_training_client.py

```diff
@@ -0,0 +1,121 @@
+from typing import Mapping, List, Optional
+
+from grpclib.client import Channel
+
+from viam import logging
+from viam.proto.app.mltraining import (
+    CancelTrainingJobRequest,
+    GetTrainingJobRequest,
+    GetTrainingJobResponse,
+    ListTrainingJobsRequest,
+    ListTrainingJobsResponse,
+    MLTrainingServiceStub,
+    ModelType,
+    SubmitTrainingJobRequest,
+    SubmitTrainingJobResponse,
+    TrainingStatus,
+    TrainingJobMetadata,
+)
+from viam.proto.app.data import (
+    Filter,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+class MLTrainingClient:
+    """gRPC client for working with ML training jobs.
+
+    Constructor is used by `ViamClient` to instantiate relevant service stubs.
+    Calls to `MLTrainingClient` methods should be made through `ViamClient`.
+    """
+
+    def __init__(self, channel: Channel, metadata: Mapping[str, str]):
+        """Create a `MLTrainingClient` that maintains a connection to app.
+
+        Args:
+            channel (grpclib.client.Channel): Connection to app.
+            metadata (Mapping[str, str]): Required authorization token to send requests to app.
+        """
+        self._metadata = metadata
+        self._ml_training_client = MLTrainingServiceStub(channel)
+        self._channel = channel
+
+    async def submit_training_job(
+        self,
+        org_id: str,
+        model_name: str,
+        model_version: str,
+        model_type: ModelType,
+        tags: List[str],
+        filter: Optional[Filter] = None,
+    ) -> str:
+        """Submits a training job to be processed by the training job manager.
+
+        Args:
+            org_id (str): the ID of the organization whose data will be used for training.
+            model_name (str): the training model name.
+            model_version (str): the training model version.
+            model_type (viam.proto.app.mltraining.ModelType): the training model type.
+            tags (List[str]): list of tags to apply to the training model output.
+            filter (Optional[viam.proto.app.data.Filter]): `Filter` specifying which data to include when training.
+            If unspecified, all data will be included.
+
+        Returns:
+            str: the id assigned to the training job.
+        """
+
+        request = SubmitTrainingJobRequest(
+            organization_id=org_id,
+            model_name=model_name,
+            model_version=model_version,
+            model_type=model_type,
+            tags=tags,
+            filter=filter,
+        )
+        response: SubmitTrainingJobResponse = await self._ml_training_client.SubmitTrainingJob(request, metadata=self._metadata)
+
+        return response.id
+
+    async def get_training_job(self, id: str) -> TrainingJobMetadata:
```

Comment:
(note to reviewers) I think there's an argument to be made for creating our own shadow type for `TrainingJobMetadata`. The type ultimately breaks down into a struct of primitives, a request (which is just a a string and a `filter`, which we already have conversion functions for), and an enum (which is relatively straightforward). So, the type isn't terribly complicated, but it is a bit nested (i.e., it contains a request which contains a filter) and I could imagine users not loving interacting with it.

I opted for the simpler thing for now because 1) why do extra work if it's not necessary, and 2) my understanding is @njooma has a preference for using the `proto` types when we can reasonably get away with it. But, I'm curious to get y'all's thoughts. You can see what a manually constructed `TrainingJobMetadata` looks like in the test file below.

Reply:
I can see how hoisting some of the fields to top-level would be useful. If we were to make a shadow class, would filter also be automatically converted to the python-native type?

Reply:
I don't think we actually have a python-native `Filter` type, we just have a helper function for creating a `Filter`. So users would still be stuck with the proto type unless we want to now create a python-native `Filter`. This would be a bit of a reversal from our previous approach, but maybe worth considering given the now-expanded presence of `Filter`s?

Reply:
In that case I think we can leave this as is -- no need to create a shadow type